core/authority-discovery/primitives/src/lib.rs

```diff
@@ -14,18 +14,19 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Common consensus primitives.
+//! Runtime Api to help discover authorities.
 
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use codec::Codec;
 use client::decl_runtime_apis;
+use codec::Codec;
 use rstd::vec::Vec;
 
 decl_runtime_apis! {
-	/// Common consensus runtime api.
-	pub trait ConsensusApi<AuthorityId: Codec> {
-		/// Returns the set of authorities of the currently active consensus mechanism.
-		fn authorities() -> Vec<AuthorityId>;
-	}
+      pub trait AuthorityDiscoveryApi<AuthorityId: Codec> {
```

Comment:
is it not redundant to call it `Api`?

Reply:
and where are the docs?

Reply:
> is it not redundant to call it Api?

I thought this was a convention for a runtime Api e.g. like `OffChainApi`, `GrandpaApi` and `AuraApi`. Also while `AuthorityDiscovery` is the entire process of making our self discoverable and discovering others, the `Api` is only one part enabling logic in `/core` to interoperate with logic in `/srml`.